packages/wallet-sdk/src/sign/SignerConfigurator.ts

```diff
@@ -41,129 +48,114 @@ export class SignerConfigurator {
-    this.initWalletLinkSigner = this.initWalletLinkSigner.bind(this);
   }
 
-  private readonly updateRelay: SignerUpdateListener = {
-    onAccountsUpdate: (signer, ...rest) => {
-      if (this.signer && signer !== this.signer) return; // ignore events from inactive signers
-      this.updateListener.onAccountsUpdate(...rest);
-    },
-    onChainUpdate: (signer, ...rest) => {
-      if (this.signer && signer !== this.signer) return; // ignore events from inactive signers
-      if (signer instanceof WLSigner) {
-        this.signerTypeSelectionResolver?.('walletlink');
-      }
-      this.updateListener.onChainUpdate(...rest);
-    },
-  };
-
-  initSigner = () => {
-    switch (this.signerType) {
-      case 'scw':
-        this.initScwSigner();
+  handleSignerConfigMessage(message: ConfigMessage) {
+    switch (message.event.type) {
+      case SignerConfigEventType.SignerTypeSelected:
+        this.resolveSelectSignerType?.(message.event.value as SignerType);
+        this.resolveSelectSignerType = undefined;
         break;
-      case 'walletlink':
-        this.initWalletLinkSigner();
-        break;
-      case 'extension':
-        this.initExtensionSigner();
+      case WalletLinkConfigEventType.RequestWalletLinkUrl:
+        this.popupCommunicator.postConfigMessage(
+          WalletLinkConfigEventType.WalletLinkUrl,
+          this.getWalletLinkQRCodeUrl()
+        );
         break;
     }
-  };
-
-  private initScwSigner() {
-    if (this.signer instanceof SCWSigner) return;
-
-    this.signer = new SCWSigner({
-      appName: this.appName,
-      appLogoUrl: this.appLogoUrl,
-      appChainIds: this.appChainIds,
-      puc: this.popupCommunicator,
-      updateListener: this.updateRelay,
-    });
   }
 
-  private initWalletLinkSigner() {
-    if (this.signer instanceof WLSigner) return;
-
-    this.signer = new WLSigner({
-      appName: this.appName,
-      appLogoUrl: this.appLogoUrl,
-      updateListener: this.updateRelay,
-    });
+  async onDisconnect() {
+    this.signerTypeStorage.removeItem(SIGNER_TYPE_KEY);
+    await this.initiatedWLSigner?.disconnect();
+    this.initiatedWLSigner = undefined;
   }
 
-  private initExtensionSigner() {
-    const extensionSigner = window.coinbaseWalletExtensionSigner;
-    if (!extensionSigner) {
-      throw standardErrors.provider.unauthorized('Coinbase Wallet extension signer not found');
-    }
+  private async selectSignerType(): Promise<SignerType> {
+    return new Promise((resolve, reject) => {
+      this.resolveSelectSignerType = (signerType: SignerType) => {
+        this.storeSignerType(signerType);
+        resolve(signerType);
+      };
+      this.popupCommunicator.rejectSignerSelection = reject;
 
-    this.signer = extensionSigner;
+      this.popupCommunicator.connect({
+        onboarding: true,
+        smartWalletOnly: this.smartWalletOnly,
+        isExtSignerAvailable: Boolean(this.injectedSigner),
+      });
```

Comment:
1. where do we use this with different options? do we set `onboarding: false` somewhere?
2. do we need all of these? `isExtSignerAvailable` can be checked on the popup itself.

Reply:
good call
1. removed `isExtSignerAvailable`, will let SCW FE to detect extension
2. i tend to keep onboarding: true in the search param tho, so it can explicitly tell FE to redirect to onboarding flow. lmk wdyt